### ISSUE TITLE
"mask_padding_with_zero" for xlnet

### ISSUE THREAD

**Issue by tbornt** [huggingface/pytorch-transformers#1022]
## ❓ Questions & Help

From the source code in [xlnet repo](https://github.com/zihangdai/xlnet/blob/master/classifier_utils.py) line113-115
I see the comment
`
 The mask has 0 for real tokens and 1 for padding tokens. Only real
 tokens are attended to.

input_mask = [0] * len(input_ids)
`

But in this repo, I found the code for generate input_mask in examples/utils_glue.py
`
input_mask = [1 if mask_padding_with_zero else 0] * len(input_ids)
`
and `mask_padding_with_zero` for xlnet and bert is all set True.

I'm confused if this is a bug.

**Comment by thomwolf** [comment]
It is not. We've added an option to input a negative mask in XLNet so it can use the same input pattern as the other models.

If you take a look at the inputs of XLNetModel [here](https://huggingface.co/pytorch-transformers/model_doc/xlnet.html#pytorch_transformers.XLNetModel), you will see both possible masks: `attention_mask` (the original XLNet mask), `input_mask` the negative we use in the SQuAD example.

**Comment by tbornt** [comment]
Oh, I see. Great work to maintain consistency with other models.
